setup.py

```diff
@@ -46,12 +46,19 @@
   keywords='appscale google-app-engine python java go php',
   platforms='Posix; MacOS X',
   install_requires=[
+    'adal',
+    'azure==2.0.0rc5',
+    'cryptography',
     'argparse',
     'boto',
     'google-api-python-client>=1.5.0',
+    'haikunator',
     'httplib2',
+    'jwt',
```

Comment:
Yes, in order to use the ADAL library, cryptography is required and a sub dependency of cryptography is jwt and libffi-dev. We don't use any of them directly, but they need to be installed before using adal so I have defined them explicitly as none of the other packages will install them on their own.